### PR TITLE
Upgrade to Cypress 12

### PR DESCRIPTION
**Pull request by rsmithadhoc**
## Summary

- Updated Cypress to v12.
  - `cy.server()` was removed from Cypress.
  - `cy.route()` replaced with `cy.intercept` in Cypress.
  - I didn't see any explicit documentation on this, but some checks using `.should('equal', undefined)` were purposefully looking for `undefined`. Cypress could not check for that element in the new version (but maybe it could in the past), which makes sense to me since it is undefined. I removed those checks for `undefined`.
  - Improper chaining [commands and actions](https://docs.cypress.io/guides/core-concepts/retry-ability#Actions-should-be-at-the-end-of-chains-not-the-middle) are reported as linting errors, I addressed those.
- Updated other Cypress packages to their latest versions.
 
**Note:** Some of these fail locally without the back-end running because it is trying to get the feature toggles instead of mocking the call. This was a pre-existing issue unrelated to this update.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/59245

## Testing done

- Tested each changed test locally after updating it.
- CI ran without issue.

### CODE REVIEW

File: src/applications/personalization/profile/tests/e2e/profile-a11y.cypress.spec.js [hunk 1]
```diff
@@ -50,9 +50,6 @@ function checkAllPages(mobile = false) {
   );
   cy.title().should('eq', 'Personal Information | Veterans Affairs');
 
-  // focus should be on the sub-nav's heading when redirected from /profile/
-  cy.focused().contains(/profile/i);
```
[the remark this answers is not in the document]
Comment (adamwhitlock1) [reply src/applications/personalization/profile/tests/e2e/profile-a11y.cypress.spec.js:54]: That code looks to be about 2 years old, so thanks for flagging that. I'll look into creating an actual assertion in that test so it checks things appropriately.

**Review by adamwhitlock1 — APPROVED** [review]
This lgtm from the Authenticated Experience side. I will make a ticket to clean up the focus test issues.

File: src/applications/personalization/profile/tests/e2e/profile-a11y.cypress.spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (randimays) [reply src/applications/personalization/profile/tests/e2e/profile-a11y.cypress.spec.js:54]: @rsmithadhoc We may need to tag the team that owns these files to address this in a separate PR

**Review by randimays — APPROVED** [review]
LGTM

File: src/applications/personalization/profile/tests/e2e/profile-a11y.cypress.spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (randimays) [reply src/applications/personalization/profile/tests/e2e/profile-a11y.cypress.spec.js:54]: I had a stale window when I came back to comment on this chain. Just saw Adam's comment. Ignore mine! :)

**Review by acrollet — APPROVED** [review]
LGTM for `coronavirus-vaccination` changes

**Review by Andrew565 — APPROVED** [review]

**Review by Mottie — APPROVED** [review]
Reviewed LGY/COE code 👍🏽

**Pull request merged**